src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -865,13 +867,43 @@ internal static string ResolvePackagePath(string searchDirectory, string searchP
             {
                 // If the search does not contain wild cards, and the target path shares the same extension, copy it
                 // e.g. <file src="ie\css\style.css" target="Content\css\ie.css" /> --> Content\css\ie.css
-                return targetPath;
+                packagePath = targetPath;
+                targetPath = null;
             }
             else
             {
                 packagePath = Path.GetFileName(fullPath);
             }
-            return Path.Combine(targetPath ?? String.Empty, packagePath);
+
+            string path = Path.Combine(targetPath ?? string.Empty, packagePath);
+
+            // Translate the tfm alias to its actual framework if necessary.
+            foreach (string knownFolder in PackagingConstants.Folders.Known)
+            {
+                string folderPrefix = knownFolder + Path.DirectorySeparatorChar;
```

Comment:
Like I wrong in another comment, in case this is comparing paths in the nupkg, using the current platform's directory separator might not work. But if it's comparing the filesystem path, it should be fine.